packages/gatsby-link/package.json

```diff
@@ -17,12 +17,12 @@
   "license": "MIT",
   "devDependencies": {
     "babel-cli": "^6.26.0",
-    "cross-env": "^5.0.5",
-    "react-router-dom": "^4.2.2"
+    "cross-env": "^5.0.5"
   },
   "dependencies": {
     "babel-runtime": "^6.26.0",
     "prop-types": "^15.5.8",
+    "react-router-dom": "^4.2.2",
```

Comment:
can this handle multiple instances? its in devDeps now because gatsby essentially provides the module itself, but I agree that is pretty tenuous

Reply:
Then `gatsby-link` should be part of `gatsby` itself instead of being packaged by itself.

Reply:
or it should be a peer dep

Reply:
IMO this isn't a peer dependency since https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-link/src/index.js#L3 directly requires stuff from `react-router-dom`.

Reply:
it also directly requires React which is a peer dep :P Sorry i'm not trying to be difficult I think it should be a direct dependency but we need to know if it is resilient to multiple instances (react-dom for instance is not which is why its a peer)

Reply:
Thats' true indeed (although React isn't even listed as peer dep) and might be tenuous as well for the same reasons mentioned above :)

Should we add both as peer dependencies?

Reply:
Honestly I hate peer dependencies cause all they seem to do is cause problems. Do they actually help anything? Was something broken by gatsby-link not having react-router-dom/react listed somewhere?

Reply:
yeah i mean there are situations where it doesn't exist, like running tests. but its depending on either being in a gatsby webpack env or npm dedupe which isn't great. I just put it in devDeps to handle the test case, all "real" usages of gatsby-link should be in the context of a webpack gatsby build/server and should have a problem since we explicitly try and resolve packages to gatsbys node_modules

I'd say avoid peerDeps here if we can, most idea ltho would be to put it in direct deps if the library can handle it

Reply:
Unable to run tests is exactly the reason why I opened that pull request. For some reason my test environment based on Jest isn't able to find `react-router-dom`:

```
 FAIL  src/components/atoms/button/button.test.js
  ● Test suite failed to run

    Cannot find module 'react-router-dom' from 'index.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:191:17)
      at Object.<anonymous> (node_modules/gatsby-link/index.js:34:23)
```

Is there any way I can make this work?

I have to add that this used to work in gatsby v1.9.18 and gatsby-link v1.6.15 but doesn't anymore in v1.9.52 / v1.6.20.

Reply:
@mfeltscher can you run find to see where `react-router-dom` is in your node_modules?

Reply:
It's in the root of `node_modules` which actually for some reasons seems to work now. Updated all the dependencies and all my tests seem to run just fine. Therefor I'm going to close this. Sorry for the confusion!

Reply:
Well thanks for proactively trying to solve things :-) I'll take a bit of confusion here and there for lots of contributors doing that!